test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/JsonHubProtocolTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Common.Tests.Internal.Protocol
+{
+    public class JsonHubProtocolTests
+    {
+        [Theory]
+        [InlineData("123", true, "Target", new object[] { 1, "Foo", 2.0f }, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"nonBlocking\":true,\"arguments\":[1,\"Foo\",2.0]}")]
+        [InlineData("123", false, "Target", new object[] { 1, "Foo", 2.0f }, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"arguments\":[1,\"Foo\",2.0]}")]
+        [InlineData("123", false, "Target", new object[] { true }, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"arguments\":[true]}")]
+        [InlineData("123", false, "Target", new object[] { null }, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"arguments\":[null]}")]
+        public async Task CanSerializeInvocationMessages(string invocationId, bool nonBlocking, string target, object[] arguments, string expectedOutput)
+        {
+            var message = new InvocationMessage(invocationId, target, arguments, nonBlocking);
+            await TestMessageOutput(expectedOutput, message);
+        }
+
+        [Theory]
+        [InlineData(false, NullValueHandling.Ignore, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"arguments\":[{\"StringProp\":\"SignalR!\",\"DoubleProp\":6.2831853071,\"IntProp\":42,\"DateTimeProp\":\"2017-04-11T00:00:00\"}]}")]
+        [InlineData(true, NullValueHandling.Ignore, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"arguments\":[{\"stringProp\":\"SignalR!\",\"doubleProp\":6.2831853071,\"intProp\":42,\"dateTimeProp\":\"2017-04-11T00:00:00\"}]}")]
+        [InlineData(false, NullValueHandling.Include, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"arguments\":[{\"StringProp\":\"SignalR!\",\"DoubleProp\":6.2831853071,\"IntProp\":42,\"DateTimeProp\":\"2017-04-11T00:00:00\",\"NullProp\":null}]}")]
+        [InlineData(true, NullValueHandling.Include, "{\"invocationId\":\"123\",\"type\":1,\"target\":\"Target\",\"arguments\":[{\"stringProp\":\"SignalR!\",\"doubleProp\":6.2831853071,\"intProp\":42,\"dateTimeProp\":\"2017-04-11T00:00:00\",\"nullProp\":null}]}")]
+        public async Task CanSerializeCustomObjectInInvocation(bool camelCase, NullValueHandling nullHandling, string expectedOutput)
+        {
+            var message = new InvocationMessage("123", "Target", new[] { new CustomObject() }, nonBlocking: false);
+
+            var payloadSerializer = new JsonSerializer();
+            if(camelCase)
+            {
+                payloadSerializer.ContractResolver = new CamelCasePropertyNamesContractResolver();
+            }
+            payloadSerializer.NullValueHandling = nullHandling;
```

Comment:
You could initialize this inline (i.e.: `var payloadSerializer = new JsonSerializer {NullValueHandling = nullHandling}`)